code/game/say.dm

```diff
@@ -147,7 +147,18 @@ GLOBAL_LIST_INIT(freqtospan, list(
 	//Speaker name
 	var/namepart
 	var/list/stored_name = list(null)
-	SEND_SIGNAL(speaker, COMSIG_MOVABLE_MESSAGE_GET_NAME_PART, stored_name, visible_name)
+
+	if(iscarbon(speaker)) //First, try to pull the modified title from a carbon's ID. This will override both visual and audible names.
+		var/mob/living/carbon/carbon_human = speaker
+		var/obj/item/id_slot = carbon_human.get_item_by_slot(ITEM_SLOT_ID)
+		if(id_slot)
+			var/obj/item/card/id/id_card = id_slot.GetID()
+			if(id_card)
+				SEND_SIGNAL(id_card, COMSIG_MOVABLE_MESSAGE_GET_HONORIFIC, stored_name, carbon_human)
```

Comment:
This signal seems a bit misleading, as it implies it's a signal sent to movables but it'd only ever sent to id cards

Maybe it should just be a proc `id_card/proc/get_honorifics()`

Reply:
I've renamed the signal to be more specific. I don't remember why I was set on using a signal here (I put up this PR like 3 months ago wtf am I doing), but I really don't want to change anything there right now.

Reply:
I think the signal is pretty meh but it can be refactored later, not the worst